ykbh/src/lib.rs

```diff
@@ -0,0 +1,267 @@
+use std::alloc::{alloc, dealloc, Layout};
+use std::convert::{TryFrom, TryInto};
+use ykpack::{
+    self, Constant, ConstantInt, IPlace, Local, LocalDecl, Statement, Terminator, TypeId,
+    UnsignedInt,
+};
+use yktrace::sir::SIR;
+
+const SUSIZE: usize = std::mem::size_of::<usize>();
+
+pub struct SIRInterpreter {
+    map: *mut u8,
+    layout: Layout,
+    locals: Vec<usize>,
+}
+
+impl SIRInterpreter {
+    pub fn new(local_decls: &Vec<LocalDecl>) -> Self {
+        // FIXME Soon this will be pre-computed and handed to us by SIR.
+        let mut size: usize = 0;
+        let mut locals = Vec::new();
+        for d in local_decls {
+            locals.push(size);
+            size += usize::try_from(SIR.ty(&d.ty).size()).unwrap();
+        }
+
+        // Allocate memory for the locals
+        let layout = Layout::from_size_align(usize::try_from(size).unwrap(), 8).unwrap();
```

Comment:
There may be a local that needs to be aligned to a bigger size. I don't know if alignment is really important for the interpreter though. If not, it should be 1 instead.

Reply:
There could definitely be locals which are aligned to arbitrary sizes, so I think we need to handle that. [In practise, not many things are aligned to `>` machine word, but still...]

Reply:
Actually, I did mean `1` here. This is a typo.

Reply:
Fixed in 01b6575.

Reply:
On second thought, I changed this back to `size_of::<usize>()`, which allows us to simplify the code that reads/writes pointers, and should also give us some speedup: f5cc2d7